pkg/aws/core.go

```diff
@@ -335,6 +343,20 @@ func (d *Driver) GetMachineStatus(ctx context.Context, req *driver.GetMachineSta
 
 	requiredInstance := instances[0]
 
+	// if SrcAnDstCheckEnabled is false then disable the SrcAndDestCheck on running NAT instance
+	if providerSpec.SrcAndDstChecksEnabled != nil && !*providerSpec.SrcAndDstChecksEnabled {
+
+		svc, err := d.createSVC(secret, providerSpec.Region)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		err = disableSrcAndDestCheck(svc, requiredInstance.InstanceId)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+	}
+
```

Comment:
Isn't it a bit weird that the `getMachineStatus` call has a side-effect and updates the instance?

Reply:
Indeed, @kon-angelo. You are correct. 

In order to modify this attribute, the AWS instance needs to be `Running`. So, if the `MachineController` fails just after the machine creation on AWS and just before the modification of `SrcAndDstCheck` attribute, then there would be no way MCM when restarted will know that this is missed. 

So, even though `GetMachineStatus()` driver method is expected to work atomically only to get the machine status, we have introduced this where it actually updates the machine attribute. However, it is covered & mentioned in another story [here](https://github.com/gardener/machine-controller-manager/issues/635) where it will be handled better in the future.